packages/wrangler/src/api/pages/deploy.tsx

```diff
@@ -360,12 +361,35 @@ export async function deploy({
 		}
 	}
 
-	const deploymentResponse = await fetchResult<Deployment>(
-		`/accounts/${accountId}/pages/projects/${projectName}/deployments`,
-		{
-			method: "POST",
-			body: formData,
+	let attempts = 0;
+	let lastErr: unknown;
+	while (attempts < MAX_DEPLOYMENT_ATTEMPTS) {
+		try {
+			const deploymentResponse = await fetchResult<Deployment>(
+				`/accounts/${accountId}/pages/projects/${projectName}/deployments`,
+				{
+					method: "POST",
+					body: formData,
+				}
+			);
+			return deploymentResponse;
+		} catch (e) {
+			lastErr = e;
+			if (
+				(e as { code: number }).code === 8000000 &&
```

Comment:
What's this error code? Can we move to constant so the intent is clearer?

Reply:
@jrf0110 this seemed like an issue greater than the scope of this PR, so am addressing in a dedicated PR here: https://github.com/cloudflare/workers-sdk/pull/3801